bokeh/models/annotations.py

```diff
@@ -439,9 +439,10 @@ class Tooltip(Annotation):
     """
     level = Override(default="overlay")
 
-    side = Either(Auto, Enum(Side), default="auto", help="""
+    attachment = Enum("horizontal", "vertical", "left", "right", "above", "below", help="""
```

Comment:
For now `"auto"` could just map to one of the others, but it leaves the door open for the future to do something a bit smarter.